tls/s2n_handshake.h

```diff
@@ -160,6 +160,15 @@ struct s2n_handshake {
 
     /* Set to 1 if the RSA verification failed */
     uint8_t rsa_failed;
+
+    /* Tracks if the client has seen a HelloRetryRequest */
+    unsigned client_received_hrr:1;
+
+    /* Tracks whether the server sent a HelloRetryRequest */
+    unsigned server_sent_hrr:1;
+
+    /* Tracks whether the server will need to send a HelloRetryRequest */
+    unsigned requires_retry:1;
```

Comment:
requires retry may hold multiple meaning, does it sound clearer if it's named `server_requires_hrr`?

Reply:
Addressed in above comment. I've renamed this and updated the comments.